modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Semiring.kt

```diff
@@ -7,16 +7,29 @@ import arrow.core.Option
  */
 interface Semiring<A> : Monoid<A> {
 
+    /**
+     * A zero value for this A
+     */
     fun zero(): A
+
+    /**
+     * A one value for this A
+     */
     fun one(): A
 
+    /**
+     * Multiplicatively combine two [A] values.
+     */
     fun A.combineMultiplicate(b: A): A
 
     override fun empty(): A = one()
 
     operator fun A.times(b: A): A =
             this.combineMultiplicate(b)
 
-    fun A.maybeProduct(b: A?): A =
+    /**
+     * Maybe multiplicatively combine two [A] values.
+     */
+    fun A.maybeCombineMultiplicate(b: A?): A =
```

Comment:
Sure, but I wanted to be consistent with Semigroup's [maybeCombine](https://github.com/arrow-kt/arrow/blob/e3591267c6718c66beaaf1a5f5c595e1905c66ae/modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Semigroup.kt#L17)

Reply:
Should I change it in this this PR for both functions, create a new PR or just leave it?

Reply:
I'd fix both on this PR tbh :P

Reply:
K :grin: But just to be sure, returning `zero()` if the receiver is null is the expected behavior, right?

Reply:
zero for addition, one for multiplication

Reply:
Mhh maybe the reason why `maybeCombine`'s  receiver is not nullable is because `maybeCombine` is defined in `Semigroup`, which doesn't have an empty element. To get around this I have added `A?.maybeCombineAddition` to `Semiring`